Fix "error: [Errno 98] Address already in use" exposed during testing

Tests have become rather flaky, due to the issue `error: [Errno 98] Address already in use` which can be viewed in the following test:

https://travis-ci.org/uber/tchannel-python/jobs/85838961

```
tests/tornado/test_connection.py ..
tests/tornado/test_dispatch.py ....
tests/tornado/test_hyperbahn.py .....
tests/tornado/test_peer.py ...........
tests/tornado/test_request.py .
tests/tornado/test_tchannel.py .....
tests/tornado/test_timeout.py .

==================================== ERRORS ====================================
__________ ERROR at setup of test_set[client_thriftrw-server_thrift] ___________

io_loop = <tornado.platform.epoll.EPollIOLoop object at 0x2aae9b7dd110>

[1m    @pytest.fixture[0m
[1m    def server(io_loop):  # need io_loop fixture for listen() to work[0m
[1m        server = TChannel(name='server')[0m
[1m>       server.listen()[0m

tests/schemes/test_thrift.py:65: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tchannel/tchannel.py:227: in listen
[1m    return self._dep_tchannel.listen(port)[0m
tchannel/tornado/tchannel.py:259: in listen
[1m    sockets = bind_sockets(self._port)[0m
.tox/py27/lib/python2.7/site-packages/tornado/netutil.py:187: in bind_sockets
[1m    sock.bind(sockaddr)[0m
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

name = 'bind', self = <socket._socketobject object at 0x2aae9b7dffa0>
args = (('0.0.0.0', 37892),)

[1m    def meth(name,self,*args):[0m
[1m>       return getattr(self._sock,name)(*args)[0m
[1m[31mE       error: [Errno 98] Address already in use[0m

/opt/python/2.7.9/lib/python2.7/socket.py:224: error
[1m[31m=============== 430 passed, 11 skipped, 1 error in 9.80 seconds ================[0m
[31mERROR: InvocationError: '/home/travis/build/uber/tchannel-python/.tox/py27/bin/py.test --cov-report=term-missing tests'[0m
___________________________________ summary ____________________________________
[31mERROR:   py27: commands failed[0m
make: *** [test_ci] Error 1
travis_time:end:11f2f308:start=1445036649983119834,finish=1445036718931727174,duration=68948607340
[0K
[31;1mThe command "make test_ci" exited with 2.[0m

Done. Your build exited with 1.
```  

This is a costly bug - let's fix it once and for all.

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -24,7 +24,7 @@ Changes by Version
   despite closing the channel.
 - Explicit destinations for ``ThriftArgScheme`` may now be specified on a
   per-request basis by using the ``hostport`` keyword argument.
-
+- Only listen on IPv4, until official IPv6 support arrives.
 
 
 0.17.11 (2015-10-19)

--- a/tchannel/tornado/tchannel.py
+++ b/tchannel/tornado/tchannel.py
@@ -273,6 +273,7 @@ class TChannel(object):
             self._port,
             # ipv6 causes random address already in use (socket.error w errno
             # == 98) when getaddrinfo() returns multiple values
+            # @see https://github.com/uber/tchannel-python/pull/257
             family=socket.AF_INET,
         )
         assert sockets, "No sockets bound for port %d" % self._port

--- a/tchannel/tornado/tchannel.py
+++ b/tchannel/tornado/tchannel.py
@@ -23,6 +23,7 @@ from __future__ import absolute_import
 import inspect
 import logging
 import os
+import socket
 import sys
 from functools import partial
 
@@ -268,7 +269,12 @@ class TChannel(object):
         assert self._handler, "Call .host with a RequestHandler first"
         server = TChannelServer(self)
 
-        sockets = bind_sockets(self._port)
+        sockets = bind_sockets(
+            self._port,
+            # ipv6 causes random address already in use (socket.error w errno
+            # == 98) when getaddrinfo() returns multiple values
+            family=socket.AF_INET,
+        )
         assert sockets, "No sockets bound for port %d" % self._port
 
         # If port was 0, the OS probably assigned something better.

--- a/tchannel/tornado/tchannel.py
+++ b/tchannel/tornado/tchannel.py
@@ -273,7 +273,7 @@ class TChannel(object):
             self._port,
             # ipv6 causes random address already in use (socket.error w errno
             # == 98) when getaddrinfo() returns multiple values
-            # @see https://github.com/uber/tchannel-python/pull/257
+            # @see https://github.com/uber/tchannel-python/issues/256
             family=socket.AF_INET,
         )
         assert sockets, "No sockets bound for port %d" % self._port

